dataflow_transfer/run_classes/generic_runs.py

```diff
@@ -0,0 +1,62 @@
+import os
+import logging
+
+from dataflow_transfer.utils.transfer import sync_to_hpc
+
+logger = logging.getLogger(__name__)
+
+
+class Run:
+    """Defines a generic sequencing run"""
+
+    def __init__(self, run_dir, configuration):
+        self.run_dir = run_dir
+        self.run_id = os.path.basename(run_dir)
+        self.configuration = configuration
+        self.final_file = ""
+        self.rsync_log = os.path.join(
+            self.run_dir, "rsync.log"
+        )  # TODO: add timestamp to log filename
+
+    def sequencing_ongoing(self):
+        final_file_path = os.path.join(self.run_dir, self.final_file)
+        if os.path.exists(final_file_path):
+            return False
+        else:
+            return False
+
+    def initiate_background_transfer(self):
+        logger.info(f"Initiating background transfer for {self.run_dir}")
+        sync_to_hpc(
+            run_path=self.run_dir,
+            destination=self.miarka_destination,
+            rsync_log=self.rsync_log,
+            background=True,
+            transfer_details=self.configuration.get("transfer_details", {}),
+        )
+
+    def initiate_final_transfer(self):
+        logger.info(f"Initiating final transfer for {self.run_dir}")
+        sync_to_hpc(
+            run_path=self.run_dir,
+            destination=self.miarka_destination,
+            rsync_log=self.rsync_log,
+            background=False,
+            transfer_details=self.configuration.get("transfer_details", {}),
+        )
+
+    def sync_metadata(self):
```

Comment:
Could you explain what the difference between metadata and stats is? Maybe it can go in a comment for the functions as well.

Reply:
Sure! So in some instances we'd only update the status fields, and in some we'd gather files and stats and upload that to the db. At first I though we were going to have two separate DBs, one for status and one for metadata, but now since we put it all in the same doc maybe it makes more sense to just have one function that updates the db?